### PR TITLE
Fix links

### PR DESCRIPTION
**Pull request by reyang**
It seems some tooling change caused the CI to fail https://github.com/open-telemetry/opentelemetry-specification/pull/2415#issuecomment-1074160369. Most likely due to the recent change in `markdown-link-check@3.10.0` https://github.com/tcort/markdown-link-check/commit/65221868bb1101bf211f9791037f304eacf908a2.

Hopefully this quick workaround would unblock the CI.

Related to https://github.com/tcort/markdown-link-check/pull/193. If the author of `markdown-link-check` agrees that it's a bug rather than by-design, we might want to simply disable it in the CI for now and wait for the fix, or pin the tool to the last-known-good version.

### CODE REVIEW

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Job still failing, looks like there are a few other dead links.

**Comment by tigrannajaryan** [comment]
@reyang can you please update the PR to fix the rest of the links? We are blocked on this, no other PRs can be merged until this is fixed.

File: specification/metrics/datamodel.md [hunk 1]
```diff
@@ -1318,6 +1318,5 @@ To convert OTLP resource attributes to Prometheus labels, string Attribute value
 
 ## Footnotes
 
-<a name="otlpdatapointfn">[1]</a>: OTLP supports data point kinds that do not
-satisfy these conditions; they are well-defined but do not support standard
-metric data transformations.
+OTLP supports data point kinds that do not satisfy these conditions; they are
```
Comment (yurishkuro) [inline specification/metrics/datamodel.md:1321]: Do you want to keep `[1]`? Because it is used as a text of the link above.

**Review by yurishkuro — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

File: specification/metrics/datamodel.md (hunk 1 above)
Comment (reyang) [reply specification/metrics/datamodel.md]: Added `[1]`.

**Pull request merged**